contentcuration/contentcuration/management/commands/fix_duplicate_assessment_items.py

```diff
@@ -0,0 +1,66 @@
+import logging as logmodule
+import time
+import uuid
+
+from django.core.management.base import BaseCommand
+from django.db import transaction
+from django.db.models import Count
+from django.db.models import F
+
+from contentcuration.models import AssessmentItem
+from contentcuration.models import ContentNode
+
+logmodule.basicConfig()
+logging = logmodule.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    def handle(self, *args, **options):
+        start = time.time()
+        # Go through nodes that have assessment items with the same assessment_id
+        nodes = ContentNode.objects.filter(kind_id='exercise') \
+            .annotate(
+                num_ids=Count('assessment_items__pk'),
+                num_assessment_ids=Count('assessment_items__assessment_id', distinct=True)
+            ).exclude(num_ids=F('num_assessment_ids'))
+
+        with transaction.atomic():
+            items_to_delete = []
+            for node in nodes:
+                channel = node.get_channel()
+
+                # Go through each node's assessment items
+                for item in node.assessment_items.all():
+                    # Handle duplicate assessment ids
+                    exclude_ids = [i.pk for i in items_to_delete] + [item.pk]
+                    other_duplicate_assessment_items = node.assessment_items.filter(assessment_id=item.assessment_id).exclude(pk__in=exclude_ids)
+
+                    if other_duplicate_assessment_items.exists():
+                        # Remove duplicates
+                        if other_duplicate_assessment_items.filter(
+                            question=item.question,
+                            answers=item.answers,
+                            hints=item.hints,
+                            raw_data=item.raw_data
+                        ).exists():
+                            items_to_delete.append(item)
+
+                        # Get new ids for non-duplicates
+                        elif channel:
+                            new_id = uuid.uuid4().hex
+                            while node.assessment_items.filter(assessment_id=new_id).exists():
+                                new_id = uuid.uuid4().hex
+                            item.assessment_id = new_id
+                            item.save()
+
+                        # Remove orphaned nodes with non-identical assessment items
+                        else:
+                            node.delete()
```

Comment:
@jayoshih Actually, on second thought, maybe we just want to ignore this case for now to be extra safe. 

Leaving an orphaned node with dupe assessment items in the db does no harm (as it will never get loaded in Studio) and could be cleaned up later, but we could have a problem if later on we found out this code deleted nodes that it shouldn't have. (I don't see how it could, but code doing something you didn't imagine it would or could do is sort of the very definition of a bug... ;)

Aside from this one concern, I think the code looks good!

Reply:
@kollivier we were hoping to add a unique constraint, which will require us to remove the duplicates

Reply:
@kollivier would not deleting those those prevent us from adding a pairwise uniqueness constraint to assessment items for `(contentnode.pk, assessment_id)`?

Reply:
I believe that would, but I can alternatively just make this an else clause https://github.com/learningequality/studio/pull/2722/files#diff-f93f323ce5119a201483b7c2927618bdceaf1755a3424a90c19382038129f251R49